sqs/src/main/scala/akka/stream/alpakka/sqs/scaladsl/SqsSource.scala

```diff
@@ -42,7 +43,13 @@ object SqsSource {
           case Some(t) => requestBuilder.visibilityTimeout(t.toSeconds.toInt).build()
         }
       }
-      .mapAsync(settings.parallelRequests)(sqsClient.receiveMessage(_).toScala)
+      .via(
+        BalancingMapAsyncUnordered[ReceiveMessageRequest, ReceiveMessageResponse](
+          settings.parallelRequests,
+          sqsClient.receiveMessage(_).toScala,
+          response => if (response.messages().isEmpty) 1 else settings.parallelRequests
+        )
+      )
```

Comment:
This changes the semantics from `mapAsync` to `mapAsyncUnordered` which means that messages received with an later `sqsClient.receiveMessage` call might overtake messages received with an earlier `sqsClient.receiveMessage` call in the stream.